frontend/src/components/pages/CampCreation/index.tsx

```diff
@@ -159,6 +165,89 @@ const CampCreationPage = (): React.ReactElement => {
+      const editCamp = await CampsAPIClient.getCampById(editCampId);
+
+      if (editCamp) {
+        setCampToEdit(editCamp); // Store the current state of the camp to edit
+
+        // Schedule Sessions need the data in a certain format. We convert the array of dates stored in the backend
+        // to this array of CreateCampSessions so that the sessions can be viewed/edited/deleted
+        const currentCampSessions: CreateCampSession[] = editCamp.campSessions.map(
+          (cs) => {
+            const dates = cs.dates
+              .map((date) => new Date(date))
+              .sort((a, b) => a.getTime() - b.getTime());
+            const currentWeekDays = getSelectedWeekDaysFromDates(dates);
+            return {
+              startDate: dates[0],
+              endDate: dates[dates.length - 1],
+              dates,
+              selectedWeekDays: currentWeekDays,
+            };
+          },
+        );
+        setScheduledSessions(currentCampSessions);
+
+        // We set the customQuestions to be the current questions stored on the backend so the admin
+        // can view and edit the questions in the ui smoothly
+        const currentFormQuestions: CreateFormQuestion[] = editCamp.formQuestions.map(
+          (fq) => {
+            return {
+              question: fq.question,
+              type: fq.type,
+              category: fq.category,
+              required: fq.required,
+              description: fq.description,
+              options: fq.options,
+            };
+          },
+        );
+        setCustomQuestions(currentFormQuestions);
+
+        // Set basic camp details
+        setCampName(editCamp.name);
+        setCampDescription(editCamp.description);
+        setDailyCampFee(editCamp.fee);
+        setStartTime(editCamp.startTime);
+        setEndTime(editCamp.endTime);
+        setAgeLower(editCamp.ageLower);
+        setAgeUpper(editCamp.ageUpper);
+        setOffersEDLP(
+          editCamp.earlyDropoff.length !== 0 ||
+            editCamp.latePickup.length !== 0,
+        );
+        setEarliestDropOffTime(editCamp.earlyDropoff);
+        setLatestPickUpTime(editCamp.latePickup);
+        setPriceEDLP(editCamp.pickupFee);
+        setAddresLine1(editCamp.location.streetAddress1);
+        setAddresLine2(editCamp.location.streetAddress2 ?? "");
+        setCity(editCamp.location.city);
+        setProvince(editCamp.location.province);
+        setPostalCode(editCamp.location.postalCode);
+        setCampImageURL(editCamp.campPhotoUrl);
+
+        // Set camp level capacity from camp sessions
+        if (editCamp.campSessions.length > 0) {
+          // At the creation stage, all the sessions should have the same capacity
+          setCampCapacity(editCamp.campSessions[0].capacity);
+        }
+      }
+    };
+
+    // We only want to fetch the current state of the camp if we are editing a draft camp
+    if (editCampId) {
+      setInitialEditState();
+    }
+  }, [editCampId]);
```

Comment:
I don't think we need the `editCampId`? 
- It should re-evaluate if the URL is changed IIRC and never change while on the same camp
- Can combine this with the other `useEffect()`

Reply:
The linter was complaining about the dependancy being missing so I added it. I don't expect this to change whilst on the same url so thought it wouldn't make a difference. As for combining the use effects, I thought it was more readable to separate them. Otherwise, the singular useEffect would be really long and hard to understand. This way, we can isolate the logic for populating the initial edit data. thoguhts?